src/ga.js

```diff
@@ -122,15 +126,13 @@ class GeneticAlgorithm {
   }
 
   static selectOne() {
-    let index = 0;
-    let r = Math.random();
-    while (r > 0) {
-      r -= globals.humans[index].fitness;
-      index += 1;
-    }
-
-    index -= 1;
-    return globals.humans[index];
+    do {
+      let r = Math.random();
+      var betterHumans = globals.humans.filter(h => h.fitness > r);
+    } while (betterHumans.length == 0);
+    
+    let one = betterHumans[Math.floor(Math.random()*betterHumans.length)];
+    return one;
```

Comment:
I think this algorithm is slower than the existing one due to array manipulations.

Any reason why you changed it ?

Reply:
@leshchenko1979 I did some tests and you were right. The selectOne() function wasn't performing very well.

Thanks for contributing :)